### PR TITLE
[Merged by Bors] - feat(ModelTheory): Turning polynomials into ring terms

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by awainverse** [comment]
I can see how this is about to be useful in defining terms in the language of rings, but there's no actual logic in this yet, as you can see from the imports. Should this be moved nearer where `FreeCommRing` is defined and worked with?

**Review by jcommelin — COMMENTED** [review]
@ChrisHughes24 What do you think of the suggestion by awainverse to move this file out of `ModelTheory/`?

File: Mathlib/ModelTheory/Algebra/Ring/MvPolynomial.lean [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    ⟨p, fun _ => Finset.Subset.refl _⟩
+
+
+@[simp]
+theorem lift_genericPolyMap [DecidableEq κ] [CommRing R]
+    [DecidableEq R] (mons : ι → Finset (κ →₀ ℕ))
+    (f : (i : ι) × { x // x ∈ mons i } ⊕ κ → R) (i : ι) :
+    FreeCommRing.lift f (genericPolyMap mons i) =
+      MvPolynomial.eval (f ∘ Sum.inr)
+        (((mvPolynomialSupportLEEquiv mons).symm
```
Comment (ChrisHughes24) [inline Mathlib/ModelTheory/Algebra/Ring/MvPolynomial.lean:79]:
```suggestion
    FreeCommRing.lift f (genericPolyMap monoms i) =
      MvPolynomial.eval (f ∘ Sum.inr)
        (((mvPolynomialSupportLEEquiv monoms).symm
```

File: Mathlib/ModelTheory/Algebra/Ring/MvPolynomial.lean [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    (p : ι → MvPolynomial κ R) : (mvPolynomialSupportLEEquiv (fun i => (p i).support)).symm
+      (fun i => (p i.1).coeff i.2.1) = ⟨p, fun _ => Finset.Subset.refl _⟩ :=
+  (mvPolynomialSupportLEEquiv (R := R) (fun i : ι => (p i).support)).symm_apply_apply
+    ⟨p, fun _ => Finset.Subset.refl _⟩
+
+
+@[simp]
+theorem lift_genericPolyMap [DecidableEq κ] [CommRing R]
+    [DecidableEq R] (mons : ι → Finset (κ →₀ ℕ))
+    (f : (i : ι) × { x // x ∈ mons i } ⊕ κ → R) (i : ι) :
```
Comment (ChrisHughes24) [inline Mathlib/ModelTheory/Algebra/Ring/MvPolynomial.lean:76]:
```suggestion
    (f : (i : ι) × { x // x ∈ monoms i } ⊕ κ → R) (i : ι) :
```

**Comment by ChrisHughes24** [comment]
> @ChrisHughes24 What do you think of the suggestion by awainverse to move this file out of `ModelTheory/`?

I think it's a good idea.

**Comment by jcommelin** [comment]
@ChrisHughes24 Just to make sure we're on the same page: you're still planning to move this out of `ModelTheory/` right?

**Comment by ChrisHughes24** [comment]
> @ChrisHughes24 Just to make sure we're on the same page: you're still planning to move this out of `ModelTheory/` right?

I've just moved it. Lean isn't super high priority right now. That's the reason for the delay.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**